File: agent-service/app/models/study_plan.py
"""
备考计划模型
"""
from sqlalchemy import Column, String, Integer, Date, TIMESTAMP, DECIMAL, SmallInteger, ForeignKey, Text
from sqlalchemy.sql import func
from sqlalchemy.orm import relationship
from app.models.database import Base
import uuid


class StudyPlan(Base):
    """备考计划表"""
    __tablename__ = "study_plans"

    plan_id = Column(String(64), primary_key=True, default=lambda: str(uuid.uuid4()))
    user_id = Column(String(64), ForeignKey("users.user_id"), nullable=False)
    exam_name = Column(String(100), nullable=False)
    exam_type = Column(String(64), nullable=False)  # 考研、期末、考证、考公、语言、期中
    current_mode = Column(SmallInteger, default=1)  # 0-静默 1-温柔 2-强化 3-唐僧
    exam_date = Column(Date, nullable=False)
    daily_hours = Column(DECIMAL(3, 1), nullable=False)
    foundation_level = Column(SmallInteger, default=1)  # 0-零基础 1-有一定基础 2-已复习一轮
    weak_subjects = Column(String(200), nullable=True)  # JSON数组
    plan_status = Column(SmallInteger, default=0)  # 0-进行中 1-已完成 2-已暂停
    created_at = Column(TIMESTAMP, server_default=func.now())
    current_phase = Column(SmallInteger, default=1)  # 1-基础 2-强化 3-冲刺

    # 关系
    user = relationship("User", backref="plans")

    def __repr__(self):
        return f"<StudyPlan(plan_id={self.plan_id}, exam_name={self.exam_name})>"

File: backend/src/main/java/com/prepkeeper/entity/StudyPlan.java
package com.prepkeeper.entity;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDate;
import java.time.LocalDateTime;

/**
 * 备考计划实体
 */
@Data
@Entity
@Table(name = "study_plans")
public class StudyPlan {

    @Id
    @Column(name = "plan_id", length = 64)
    private String planId;

    @Column(name = "user_id", length = 64, nullable = false)
    private String userId;

    @Column(name = "exam_name", length = 100, nullable = false)
    private String examName;

    @Column(name = "exam_type", length = 64, nullable = false)
    private String examType;

    @Column(name = "exam_date", nullable = false)
    private LocalDate examDate;

    @Column(name = "daily_hours", precision = 3, scale = 1)
    private BigDecimal dailyHours;

    @Column(name = "foundation_level")
    private Integer foundationLevel = 1;

    @Column(name = "weak_subjects", length = 500)
    private String weakSubjects;

    /**
     * 监督模式: 0-静默 1-温柔 2-强化 3-唐僧
     */
    @Column(name = "current_mode")
    private Integer currentMode = 1;

    /**
     * 计划状态: 0-进行中 1-已完成 2-已暂停
     */
    @Column(name = "plan_status")
    private Integer planStatus = 0;

    /**
     * 当前阶段: 1-基础 2-强化 3-冲刺
     */
    @Column(name = "current_phase")
    private Integer currentPhase = 1;

    @Column(name = "created_at")
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @PrePersist
    protected void onCreate() {
        createdAt = LocalDateTime.now();
        updatedAt = LocalDateTime.now();
    }

    @PreUpdate
    protected void onUpdate() {
        updatedAt = LocalDateTime.now();
    }
}